### ISSUE TITLE
libpcre2-10.43 broken on gcc2

### ISSUE THREAD

**Issue by Begasus** [haikuports/haikuports#10138]
Latest (at this moment) can't be build with gcc2, parse errors, I got a few fixed by lot's of others there (skill are not up to that).

**Comment by threedeyes** [comment]
After the update libpcre2 (https://github.com/haikuports/haikuports/commit/91301503c2fb033ffce416ab8af41f91b6440261), Qt applications stop working under 32 bit Haiku. The error is **invalid opcode exception**. If you roll back libpcre2 version, the problem disappears.

**Comment by Begasus** [comment]
@threedeyes is this with the new version?
I've got latest x86 (10.43 and an older one 8.45) installed and 10.39 for gcc2, Qt-Assistant for instance still launches (on R1B4 still though).

**Comment by augiedoggie** [comment]
I noticed this problem with KeePassXC last night on a fully updated R1/B4 installation.

**Comment by Begasus** [comment]
The 32bit install here is a fully synced R1B4 also here, launches fine here after installing on both 32bit and 64bit?

**Comment by Begasus** [comment]
Screenshot taken a minute ago:

![KeePassXC](https://github.com/haikuports/haikuports/assets/16057090/9c4ffe20-9d31-4af6-a057-1c682e08e520)

**Comment by Begasus** [comment]
Updated screenshot with pcre2 packages installed:

![KeePassXC](https://github.com/haikuports/haikuports/assets/16057090/b368b8fa-5c7d-4e24-bf18-1136a357da53)

**Comment by threedeyes** [comment]
> The 32bit install here is a fully synced R1B4 also here, launches fine here after installing on both 32bit and 64bit?

The problem is reproduced on a fully updated R1B4 32bit. When running any applications using libpcre2 (including Qt applications) I get invalid opcode exception.  Also apparently this package failed for the same reason - https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/5789/builds/82857.log

The error code says, the problem may be related to the processor type and appear on different same. Perhaps some optimizations were included when building the new version of libpcre2?

PS: I have AMD Rizen 7.

**Comment by augiedoggie** [comment]
I attempted to rebuild 10.43 in my VM and run the test suite with `hp --test libpcre2_x86` and it continuously crashed with invalid opcodes.  I'm on a Ryzen 5.

**Comment by Begasus** [comment]
Intel I5 here, will do testbuilds on 32bit with 10.39 and 10.43 to build kdoctools, if it fails on the newer one I'll disable the newer one untill this is fixed (or either of you can if you beat me to it).

**Comment by Begasus** [comment]
Changes done from 10.42 to 10.43 shown here: https://fossies.org/diffs/pcre2/10.42_vs_10.43/
Maybe something broke it for us, PR to bring back 10.42 is done, please check.

**Comment by threedeyes** [comment]
> Changes done from 10.42 to 10.43 shown here: https://fossies.org/diffs/pcre2/10.42_vs_10.43/
> Maybe something broke it for us, PR to bring back 10.42 is done, please check.

I see added autodetecting cpu code in 10.43

**Comment by Begasus** [comment]
> I see added autodetecting cpu code in 10.43

Right, mentioned this earlier in IRC, probably explains different errors you see there that I don't see here.

**Comment by threedeyes** [comment]
> > I see added autodetecting cpu code in 10.43
> 
> Right, mentioned this earlier in IRC, probably explains different errors you see there that I don't see here.

I found the problem - AVX2 instructions. The 32bit version of Haiku does not support them. If you disable them, everything works fine.


https://github.com/PCRE2Project/pcre2/blob/master/src/sljit/sljitNativeX86_common.c#L511

**Comment by Begasus** [comment]
Eeps, just merged to revert to 10.42, will do some testruns to disable that (or you could take care of that), in the meantime checked a build for kdoctools 5.115.0 on 32bit, didn't error out here.

**Comment by Begasus** [comment]
@threedeyes maybe you could push kdoctools again for the buildmasters, libpcre2 10.42 is ready there, if that works well we can wait with bumping/fixing libpcre2 10.43 untill you are done for the KDE updates.

**Comment by threedeyes** [comment]
> @threedeyes maybe you could push kdoctools again for the buildmasters, libpcre2 10.42 is ready there, if that works well we can wait with bumping/fixing libpcre2 10.43 untill you are done for the KDE updates.

I tested 10.43 with AVX2 disabled - kconfig and kdoctools compiled successfully.

**Comment by Begasus** [comment]
OK, then you could push the change to get 10.43 back in the depot :)

**Comment by Begasus** [comment]
Right, and to be sure the local packages get the correct version installed. +1
